lib/batch-worker-transformer.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const { Transformer } = require("@nui/asset-compute-pipeline");
+
+class BatchWorkerCallbackTransformer extends Transformer {
+
+    /**
+     * Construct a transformer for batch workers
+     * @param {*} callback 
+     * @param {*} manifest worker manifest (only one, not a list)
+     */
+    constructor(callback, manifest) {
+        super(`batch-workerCallback-${manifest.name}`, manifest);
+
+        if(this.manifest.inputs.hasBatchMode === true || this.manifest.inputs.hasBatchMode === true){
```

Comment:
is this type to check same condition twice ?

Reply:
Ah typo. I wanted to check 2 locations, manifest root and inputs.